r/R/dplyr-summarize.R

```diff
@@ -28,14 +28,107 @@ summarise.arrow_dplyr_query <- function(.data, ..., .engine = c("arrow", "duckdb
     dplyr::group_vars(.data) # vars needed for grouping
   ))
   .data <- dplyr::select(.data, vars_to_keep)
-
   if (match.arg(.engine) == "duckdb") {
     dplyr::summarise(to_duckdb(.data), ...)
-  } else {
-    if (query_on_dataset(.data)) {
-      not_implemented_for_dataset("summarize()")
+  } else if (isTRUE(getOption("arrow.summarize", FALSE))) {
```

Comment:
Is this `arrow.summarize` option intended for development/test purposes only? Do you intend to remove it before the 6.0.0 release? If so we should open a Jira for it with priority = blocker.

Reply:
Also it might be best to standardize on the `summarise` spelling with an `s`, for consistency with dplyr options like `dplyr.summarise.inform`

Reply:
I removed the flag now. It had the side effect of raising the warning you asked about below in the unsupported cases. This is probably the right path forward.